core/src/Kokkos_ExecPolicy.hpp

```diff
@@ -773,6 +773,105 @@ struct ThreadVectorRangeBoundariesStruct {
       : start(static_cast<index_type>(arg_begin)), end(arg_end) {}
 };
 
+template <Kokkos::Iterate Direction, size_t Rank, typename iType,
+          typename TeamMemberType>
+struct MDTeamThreadRangeBoundariesStruct {
+  static_assert(2 <= Rank, "Rank must be at least 2");
+  static_assert(Rank <= 8, "Rank must be at most 8");
+  static_assert(Direction == Kokkos::Iterate::Left ||
+                    Direction == Kokkos::Iterate::Right,
+                "Direction must be Left or Right");
+
+  static constexpr Kokkos::Iterate direction = Direction;
+  static constexpr size_t rank               = Rank;
+  using index_type                           = iType;
+  using team_member_type                     = TeamMemberType;
+
+  template <typename... Ns>
+  KOKKOS_INLINE_FUNCTION constexpr explicit MDTeamThreadRangeBoundariesStruct(
+      TeamMemberType const& member, Ns&&... ns)
+      : thread(member), threadDims{static_cast<iType>(ns)...} {
+    static_assert(sizeof...(ns) == Rank, "Number of ns must equal Rank");
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  MDTeamThreadRangeBoundariesStruct(TeamMemberType const& member,
+                                    const iType (&array)[Rank])
+      : thread(member) {
+    std::copy(&array[0], &array[Rank], &threadDims[0]);
+  }
+
```

Comment:
As discussed above, you wanted to remove this constructor.
```suggestion
```

Reply:
Yep, I just removed this. Thanks for reminding.